searchlib/src/vespa/searchlib/transactionlog/domainpart.cpp

```diff
@@ -527,121 +516,84 @@ DomainPart::visit(SerialNumRange &r, Packet &packet)
+    _writtenSerial = chunk.range().to();
+    _byteSize.store(lastKnownGoodPos + os.size(), std::memory_order_release);
 }
 
 bool
-DomainPart::read(FastOS_FileInterface &file,
-                 Packet::Entry &entry,
-                 Alloc & buf,
-                 bool allowTruncate)
+DomainPart::read(FastOS_FileInterface &file, IChunk::UP & chunk, Alloc & buf, bool allowTruncate)
 {
-    bool retval(true);
     char tmp[5];
     int64_t lastKnownGoodPos(file.GetPosition());
     size_t rlen = file.Read(tmp, sizeof(tmp));
     nbostream his(tmp, sizeof(tmp));
-    uint8_t version(-1);
+    uint8_t encoding(-1);
     uint32_t len(0);
-    his >> version >> len;
-    if ((retval = (rlen == sizeof(tmp)))) {
-        if ( ! (retval = (version == Encoding::Crc::ccitt_crc32) || version == Encoding::Crc::xxh64)) {
-            string msg(make_string("Version mismatch. Expected 'ccitt_crc32=1' or 'xxh64=2',"
-                                   " got %d from '%s' at position %" PRId64,
-                                   version, file.GetFileName(), lastKnownGoodPos));
-            if ((version == 0) && (len == 0) && tailOfFileIsZero(file, lastKnownGoodPos)) {
-                LOG(warning, "%s", msg.c_str());
-                return handleReadError("packet version", file, sizeof(tmp), rlen, lastKnownGoodPos, allowTruncate);
-            } else {
-                throw runtime_error(msg);
-            }
-        }
-        if (len > buf.size()) {
-            Alloc::alloc(len).swap(buf);
-        }
-        rlen = file.Read(buf.get(), len);
-        retval = rlen == len;
-        if (!retval) {
-            retval = handleReadError("packet blob", file, len, rlen, lastKnownGoodPos, allowTruncate);
-        } else {
-            nbostream_longlivedbuf is(buf.get(), len);
-            entry.deserialize(is);
-            int32_t crc(0);
-            is >> crc;
-            int32_t crcVerify(calcCrc(Encoding(version).getCrc(), buf.get(), len - sizeof(crc)));
-            if (crc != crcVerify) {
-                throw runtime_error(make_string("Got bad crc for packet from '%s' (len pos=%" PRId64 ", len=%d) : crcVerify = %d, expected %d",
-                                                file.GetFileName(), file.GetPosition() - len - sizeof(len),
-                                                static_cast<int>(len), static_cast<int>(crcVerify), static_cast<int>(crc)));
-            }
-        }
-    } else {
-        if (rlen == 0) {
-           // Eof
+    his >> encoding >> len;
+    if (rlen != sizeof(tmp)) {
+        return (rlen == 0)
+               ? true
+               : handleReadError("packet length", file, sizeof(len), rlen, lastKnownGoodPos, allowTruncate);
+    }
+
+    try {
+        chunk = IChunk::create(encoding);
+    } catch (const std::exception & e) {
+        string msg(make_string("Version mismatch. Expected 'ccitt_crc32=1' or 'xxh64=2',"
+                               " got %d from '%s' at position %ld",
```

Comment:
PRId64 is the correct format string for position.

Reply:
Fixed